docs/ingestion/automatic-compaction.md

```diff
@@ -0,0 +1,198 @@
+
+
+### Set frequency of compaction runs
+
+If you want the Coordinator to check for compaction more frequently than its indexing period, create a separate duty group and set its time period in the `coordinator/runtime.properties` file. For example, to change the auto-compaction period to 1 minute:
+```
+druid.coordinator.dutyGroups=["compaction"]
+druid.coordinator.compaction.duties=["compactSegments"]
+druid.coordinator.compaction.period=PT60S
+```
+
+## View automatic compaction statistics
+
+After the Coordinator has initiated auto-compaction, you can view compaction statistics for the datasource, including the number of bytes, segments, and intervals already compacted and those awaiting compaction. The Coordinator also reports the total bytes, segments, and intervals not eligible for compaction in accordance with its [segment search policy](../design/coordinator.md#segment-search-policy-in-automatic-compaction).
+
+In the Druid console, the Datasources view displays auto-compaction statistics. The Tasks view shows the task information for compaction tasks that were triggered by the automatic compaction system.
+
+To get statistics by API, send a [`GET` request](../operations/api-reference.md#get-10) to `/druid/coordinator/v1/compaction/status`. To filter the results to a particular datasource, pass the datasource name as a query parameter to the request—for example, `/druid/coordinator/v1/compaction/status?dataSource=wikipedia`.
+
+## Examples
+
+The following examples demonstrate potential use cases in which auto-compaction may improve your Druid performance. See more details in [Compaction strategies](../ingestion/compaction.md#compaction-strategies). The examples in this section do not change the underlying data.
+
+### Change segment granularity
+
+You have a stream set up to ingest data with `HOUR` segment granularity into the `wikistream` datasource. You notice that your Druid segments are smaller than the [recommended segment size](../operations/segment-optimization.md) of 5 million rows per segment. You wish to automatically compact segments to `DAY` granularity while leaving the latest week of data _not_ compacted as your stream consistently has data over that time period coming in.
+
+The following auto-compaction configuration compacts existing `HOUR` segments into `DAY` segments while leaving the latest week of data not compacted:
+
+```json
+{
+  "dataSource": "wikistream",
+  "granularitySpec": {
+    "segmentGranularity": "DAY"
+  },
+  "skipOffsetFromLatest": "P1W",
+}
+```
+`skipOffsetFromLatest` applies to cases in which your stream consistently ingests late arriving data. If your stream only occasionally ingests late arriving data, the auto-compaction system robustly compacts your data even without `skipOffsetFromLatest`.
```

Comment:
nit: This makes it sound like the property only applies if the stream has late arriving data. But I think the point you're trying to make here is that if. you have data that arrives latte infrequently, you do not need to be as aggressive with this setting.

```suggestion
To set choose `skipOffsetFromLatest`, consider how frequently you expect  the stream to be receiving late arriving data. If your stream only occasionally receives late arriving data, the auto-compaction system robustly compacts your data even though data is being ingested outside the `skipOffsetFromLatest` window. For most realtime streaming ingestion use cases, a setting of a few hours or a day would be reasonable.
```